byte-buddy-dep/src/test/java/net/bytebuddy/implementation/auxiliary/AbstractMethodCallProxyTest.java

```diff
@@ -55,13 +60,34 @@ protected Class<?> proxyOnlyDeclaredMethodOf(Class<?> proxyTarget) throws Except
         assertThat(auxiliaryType.getDeclaredConstructors().length, is(1));
         assertThat(auxiliaryType.getDeclaredMethods().length, is(2));
         assertThat(auxiliaryType.getDeclaredFields().length, is(proxyMethod.getParameters().size() + (proxyMethod.isStatic() ? 0 : 1)));
-        int fieldIndex = 0;
-        if (!proxyMethod.isStatic()) {
-            assertThat(auxiliaryType.getDeclaredFields()[fieldIndex++].getType(), CoreMatchers.<Class<?>>is(proxyTarget));
+        if(!proxyMethod.isStatic()){
+            for (Field field : auxiliaryType.getDeclaredFields()){
+                if (field.getType() == proxyTarget){
+                	assertThat(field.getType(), CoreMatchers.<Class<?>>is(proxyTarget));
+                    break;
+                }  
+            }
         }
-        for (Class<?> parameterType : proxyTarget.getDeclaredMethods()[0].getParameterTypes()) {
-            assertThat(auxiliaryType.getDeclaredFields()[fieldIndex++].getType(), CoreMatchers.<Class<?>>is(parameterType));
+        ArrayList<Class<?>> filteredFields = new ArrayList<Class<?>>();
+        for(Field field : auxiliaryType.getDeclaredFields()) {
+        	  if (field.getType() != proxyTarget){
+        		  filteredFields.add(field.getType());
+              }  
         }
+        Collections.sort(filteredFields, new Comparator<Class<?>>() {
+            @Override
+            public int compare(Class<?> class1, Class<?> class2) {
+                return class1.getSimpleName().compareTo(class2.getSimpleName());
+            }
+        });
+        ArrayList<Class<?>> parameterTypes = new ArrayList<Class<?>>(Arrays.asList(proxyTarget.getDeclaredMethods()[0].getParameterTypes()));
+        Collections.sort(parameterTypes, new Comparator<Class<?>>() {
```

Comment:
Instead of creating the comparator on this line and on Line 80, can you just create it once and have both this line and Line 80 use the same comparator?